gcc/rust/ast/rust-item.h

```diff
@@ -1013,11 +1015,12 @@ class Module : public VisItem
   // Unloaded module constructor
   Module (Identifier module_name, Visibility visibility,
 	  std::vector<Attribute> outer_attrs, Location locus,
-	  std::string outer_filename)
+	  std::string outer_filename, std::vector<std::string> module_scope)
     : VisItem (std::move (visibility), std::move (outer_attrs)),
       module_name (module_name), locus (locus), kind (ModuleKind::UNLOADED),
       outer_filename (outer_filename), inner_attrs (std::vector<Attribute> ()),
-      items (std::vector<std::unique_ptr<Item>> ())
+      items (std::vector<std::unique_ptr<Item>> ()),
+      module_scope (std::move (module_scope))
```

Comment:
Isn't it undefined behavior to move the parser's `inline_module_stack` vector every time we create a new Module? I'm not super clear on that aspect of C++, but I don't know what moving the vector and then emplacing_back a new name will produce.

I think it would be really nice to have a test proceeding to multiple module scope inclusions for this. Something like
```rust
mod foo {
    mod bar {
        mod baz;
    }
}

mod fooz {
    mod barz;
}
```
and see what asan/valgrind tells us. 

I might be worrying over nothing and it's quite late so feel free to disregard :)

Reply:
It is copied in at the call site (as needed). It is needed here, so it is not moved then. I'll add this test case (as well as some `#[path]` ones).

Reply:
Oh gotcha. Thanks!